src/pip/_internal/utils/misc.py

```diff
@@ -510,6 +510,14 @@ def unzip_file(filename, location, flatten=True):
                 fn = split_leading_dir(name)[1]
             fn = os.path.join(location, fn)
             dir = os.path.dirname(fn)
+            # https://github.com/pypa/pip/issues/3907
+            if os.path.commonprefix(
```

Comment:
Instead of a multi-line conditional that is duplicated, I'd prefer we add a small helper that can be independently tested too:

```py
def _located_in_directory(directory, target):
    abs_target = os.path.abspath(target)
    abs_directory = os.path.abspath(directory)

    prefix = os.path.commonprefix([abs_directory, abs_target])
    return prefix == abs_directory
```

Reply:
changes added. please review